src/osd/osd_types.cc

```diff
@@ -7399,6 +7399,10 @@ void init_pg_ondisk(
 
   ghobject_t pgmeta_oid(pgid.make_pgmeta_oid());
   t.touch(coll, pgmeta_oid);
+  if (pool->is_crimson()) {
+    t.set_alloc_hint(coll, pgmeta_oid,
+      0, 0, CEPH_OSD_ALLOC_HINT_FLAG_LOG);
```

Comment:
Is it confirmed that the rados object `pgmeta_oid`'s omap **only** stores PG-LOG specific key-values?

Reply:
In `PG:prepare_wirte()`, `pgmeta_oid `is only used for storing `PGLOG`-related information

```
    pglog.write_log_and_missing(
      t, &km, coll_ref->get_cid(), pgmeta_oid,
      peering_state.get_pgpool().info.require_rollback());
    if (!km.empty()) {
      t.omap_setkeys(coll_ref->get_cid(), pgmeta_oid, km);
    }
    if (!key_to_remove.empty()) {
      t.omap_rmkey(coll_ref->get_cid(), pgmeta_oid, key_to_remove);
    }
```


If it takes too long (as it involves classic), probably worth considering to split out the pure cleanups in "seastore.cc" around omap and merge it first?

I suppose it can minimize rebase efforts and also helps to make the actual changes clearer.

> Will do

Reply:
@cyx1231st I tried to split this PR into two PRs, including pure cleanups, but I found most of the commits depend on omap_tree changes. So, updating this PR seems to be the easier approach. What do you think?

Reply:
@myoungwon No worries, I'll take the cleanup in my TODO list (in a lower priority) in case this PR cannot be merged in time.

I suppose the consistent `omaptree_*` interfaces can make it easier to consume the other changes (like the laddr hint redesign) as well.

Reply:
Can we move this bit to crimson's ShardServices::handle_pg_create_info?

If so, let's separate this commit into:
1)  common, include: introduce LOG alloc hint

2) crimson/osd/shard_services: use set_alloc_hint when creating an pgmeta_oid for PGLog

Reply:
`sel_alloc_hint`should be called  right after `touch` because seastore needs to create a separated data structure (log_root) for LOG before handling other operations (e.g., `omap_setkeys`). In this `init_pg_ondisk` function, `omap_setkeys ` is called at the end, so that its output is stored in the original data structure (omap_root) instead of the separated data structure (log_root).

Reply:
> Is it confirmed that the rados object `pgmeta_oid`'s omap **only** stores PG-LOG specific key-values?

It seems that we also set keys such as `infover_key` and `epoch_key` to `pgmeta_oid`'s omap
```
// prefix pgmeta_oid keys with _ so that PGLog::read_log_and_missing() can
// easily skip them
static const std::string_view infover_key = "_infover";
static const std::string_view info_key = "_info";
static const std::string_view biginfo_key = "_biginfo";
static const std::string_view epoch_key = "_epoch";
static const std::string_view fastinfo_key = "_fastinfo";
```